`builtin_zstd=no` build on Fedora 43 fails following delta encoding PCK merge

### Tested versions

- Reproducible in: 4.6.dev https://github.com/godotengine/godot/commit/1667e6fcceba0908f30dc21e0aec4b643a71dfaa
- Not reproducible in: 4.5.1.stable, 4.6.dev4

### System information

Godot v4.6.dev (1667e6fcceba0908f30dc21e0aec4b643a71dfaa) - Fedora Linux 43 (KDE Plasma Desktop Edition) on X11 - X11 display driver, Multi-window, 1 monitor - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 5090 (nvidia; 580.95.05) - AMD Ryzen 9 9950X3D 16-Core Processor (32 threads) - 62.33 GiB memory

### Issue description

Following the merge of https://github.com/godotengine/godot/pull/112011, I get this build error with `builtin_zstd=no` on Fedora 43:

```
./core/io/delta_encoding.cpp: In static member function 'static Error DeltaEncoding::encode_delta(Span<unsigned char>, Span<unsigned char>, Vector<unsigned char>&, int)':
./core/io/delta_encoding.cpp:66:9: error: 'ZSTD_parameters' was not declared in this scope; did you mean 'ZSTD_dParameter'?
   66 |         ZSTD_parameters zstd_params = ZSTD_getParams(p_compression_level, p_new_data.size(), p_old_data.size());
      |         ^~~~~~~~~~~~~~~
      |         ZSTD_dParameter
./core/io/delta_encoding.cpp:67:9: error: 'zstd_params' was not declared in this scope
   67 |         zstd_params.fParams.contentSizeFlag = 1;
      |         ^~~~~~~~~~~
./core/io/delta_encoding.cpp:70:23: error: 'ZSTD_CCtx_setParams' was not declared in this scope; did you mean 'ZSTD_CCtx_setParameter'?
   70 |         zstd_result = ZSTD_CCtx_setParams(zstd_context, zstd_params);
      |                       ^~~~~~~~~~~~~~~~~~~
      |                       ZSTD_CCtx_setParameter
./core/io/delta_encoding.cpp: In static member function 'static Error DeltaEncoding::decode_delta(Span<unsigned char>, Span<unsigned char>, Vector<unsigned char>&)':
./core/io/delta_encoding.cpp:94:30: error: 'ZSTD_findDecompressedSize' was not declared in this scope; did you mean 'ZSTD_getDecompressedSize'?
   94 |         size_t zstd_result = ZSTD_findDecompressedSize(p_delta.ptr() + DELTA_HEADER_SIZE, p_delta.size() - DELTA_HEADER_SIZE);
      |                              ^~~~~~~~~~~~~~~~~~~~~~~~~
      |                              ZSTD_getDecompressedSize
```

**Edit:** It turns out it's not related to `scu_build=all`, but `builtin_zstd=no`. The version in Fedora 43 repositories (1.5.7) is the latest still, so I don't know what could cause that.

cc @mihe 

### Steps to reproduce

- Try to compile Godot with `scons builtin_zstd=no` on Fedora 43.

### Minimal reproduction project (MRP)

N/A

--- a/core/io/delta_encoding.cpp
+++ b/core/io/delta_encoding.cpp
@@ -63,15 +63,20 @@ Error DeltaEncoding::encode_delta(Span<uint8_t> p_old_data, Span<uint8_t> p_new_
 
 	ZstdCompressionContext zstd_context;
 
-	ZSTD_parameters zstd_params = ZSTD_getParams(p_compression_level, p_new_data.size(), p_old_data.size());
-	zstd_params.fParams.contentSizeFlag = 1;
-	zstd_params.fParams.checksumFlag = 1;
+	zstd_result = ZSTD_CCtx_setParameter(zstd_context, ZSTD_c_compressionLevel, p_compression_level);
+	ERR_FAIL_ZSTD_V_MSG(zstd_result, FAILED, "Failed to encode delta. Setting compression level failed.");
 
-	zstd_result = ZSTD_CCtx_setParams(zstd_context, zstd_params);
-	ERR_FAIL_ZSTD_V_MSG(zstd_result, FAILED, "Failed to encode delta. Setting compression parameters failed.");
+	zstd_result = ZSTD_CCtx_setPledgedSrcSize(zstd_context, p_new_data.size());
+	ERR_FAIL_ZSTD_V_MSG(zstd_result, FAILED, "Failed to encode delta. Setting pledged source size failed.");
 
 	zstd_result = ZSTD_CCtx_refPrefix(zstd_context, p_old_data.ptr(), p_old_data.size());
 	ERR_FAIL_ZSTD_V_MSG(zstd_result, FAILED, "Failed to encode delta. Setting prefix dictionary failed.");
+
+	zstd_result = ZSTD_CCtx_setParameter(zstd_context, ZSTD_c_contentSizeFlag, 1);
+	ERR_FAIL_ZSTD_V_MSG(zstd_result, FAILED, "Failed to encode delta. Setting content size flag failed.");
+
+	zstd_result = ZSTD_CCtx_setParameter(zstd_context, ZSTD_c_checksumFlag, 1);
+	ERR_FAIL_ZSTD_V_MSG(zstd_result, FAILED, "Failed to encode delta. Setting checksum flag failed.");
 
 	zstd_result = ZSTD_compress2(zstd_context, r_delta.ptrw() + DELTA_HEADER_SIZE, r_delta.size() - DELTA_HEADER_SIZE, p_new_data.ptr(), p_new_data.size());
 	ERR_FAIL_ZSTD_V_MSG(zstd_result, FAILED, "Failed to encode delta. Compression failed.");
@@ -91,7 +96,7 @@ Error DeltaEncoding::decode_delta(Span<uint8_t> p_old_data, Span<uint8_t> p_delt
 	ERR_FAIL_COND_V_MSG(memcmp(magic, DELTA_MAGIC, 4) != 0, ERR_FILE_CORRUPT, "Failed to decode delta. Header is invalid.");
 	ERR_FAIL_COND_V_MSG(version != DELTA_VERSION_NUMBER, ERR_FILE_UNRECOGNIZED, vformat("Failed to decode delta. Expected version %d but found %d.", DELTA_VERSION_NUMBER, version));
 
-	size_t zstd_result = ZSTD_findDecompressedSize(p_delta.ptr() + DELTA_HEADER_SIZE, p_delta.size() - DELTA_HEADER_SIZE);
+	size_t zstd_result = ZSTD_getFrameContentSize(p_delta.ptr() + DELTA_HEADER_SIZE, p_delta.size() - DELTA_HEADER_SIZE);
 	ERR_FAIL_ZSTD_V_MSG(zstd_result, FAILED, "Failed to decode delta. Unable to find decompressed size.");
 
 	r_new_data.reserve_exact(zstd_result);

